### ISSUE TITLE
How to disable the default AWS credentials profile file check in `InitAPI()`

### ISSUE THREAD

**Issue by miluoalbert** [aws/aws-sdk-cpp#1411]
Confirm by changing [ ] to [x] below:
- [ x ] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/welcome.html) and [API reference](http://sdk.amazonaws.com/cpp/api/LATEST/index.html)
- [ x ] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-cpp/issues) and didn't find any solution

**Platform/OS/Hardware/Device**
CentOS/ x86 64

**Describe the question**
I set AWS credentials by providing it to an AWS client class constructor:
```c++
s3_client = std::shared_ptr<Aws::S3::S3Client>(
        new Aws::S3::S3Client(
            Aws::Auth::AWSCredentials("myacckeyid", "mysecretkey"), cfg,
            Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never, true),
        ShutdownClient)
```
But each time when my program starts and run into:
```c++
    Aws::SDKOptions sdkoptions;
    Aws::InitAPI(sdkoptions);
```
It output several logs which shows SDK can not open the default config files `.aws/config` and `.aws/credentials`. As I don't want to provide AWS Credentials through config files or environmental variability, Is there any way to avoid the `InitAPI()` try to get configs from those default config files?

**Logs/output**
```bash
[INFO] 2020-06-28 07:28:36.225 Aws_Init_Cleanup [140296948209600] Initiate AWS SDK for C++ with Version:1.7.336
[INFO] 2020-06-28 07:28:36.226 Aws::Config::AWSConfigFileProfileConfigLoader [140296948209600] Initializing config loader against fileName /root/.aws/credentials and using profilePrefix = 0
[INFO] 2020-06-28 07:28:36.226 Aws::Config::AWSConfigFileProfileConfigLoader [140296948209600] Initializing config loader against fileName /root/.aws/config and using profilePrefix = 1
[INFO] 2020-06-28 07:28:36.227 Aws::Config::AWSConfigFileProfileConfigLoader [140296948209600] Unable to open config file /root/.aws/credentials for reading.
[INFO] 2020-06-28 07:28:36.227 Aws::Config::AWSProfileConfigLoader [140296948209600] Failed to reload configuration.
[INFO] 2020-06-28 07:28:36.227 Aws::Config::AWSConfigFileProfileConfigLoader [140296948209600] Unable to open config file /root/.aws/config for reading.
[INFO] 2020-06-28 07:28:36.227 Aws::Config::AWSProfileConfigLoader [140296948209600] Failed to reload configuration.
[INFO] 2020-06-28 07:28:36.229 CurlHttpClient [140296948209600] Initializing Curl library with version: 7.69.1, ssl version:
```

**Comment by KaibaLopez** [comment]
Hi @miluoalbert,
So tl;dr: There isn't at the moment...
To clarify though, it is not looking for the credentials, but for the client configuration, more specifically the region.

The ClientConfig has 2 constructor options, one of them allows you to directly tell it what profile to look for on the config file (by default `.aws/config`, and the default one that goes through a number of locations (including the config file) looking for the correct region. And when you create a client and don't specify configuration then it builds one using the default. 
So yea, let me know if you have any other question on this.

**Comment by miluoalbert** [comment]
Hi @KaibaLopez ,
Thanks for the response. As for my case, client info needs to be given dynamically when s3 client is constructed instead of setting by files, I'll just ignore those failed info.

Thanks a lot!

**Comment by asod1999** [comment]
How did you that?

> Hi @KaibaLopez , Thanks for the response. As for my case, client info needs to be given dynamically when s3 client is constructed instead of setting by files, I'll just ignore those failed info.
> 
> Thanks a lot!
